### PR TITLE
[FLINK-22515][docs] Add documentation for GSR-Flink Integration

### PR DESCRIPTION
**Pull request by LinyuYao1021**
## What is the purpose of the change

This pull request adds documentation for Glue Schema Registry(GSR)-Flink integration and updates with the latest GSR version.

## Brief change log

  - Documentation of Flink-Kafka and Flink-Kinesis is updated to introduce GSR integration
  - GSR version is updated to 1.0.2
  - Using `avro` plugin to converter `.asvc` file to Java POJO class.

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by LinyuYao1021** [comment]
@flinkbot run azure

**Comment by LinyuYao1021** [comment]
@flinkbot run azure

**Comment by LinyuYao1021** [comment]
@flinkbot run travis

**Comment by LinyuYao1021** [comment]
@flinkbot run travis

**Comment by LinyuYao1021** [comment]
@flinkbot run azure

File: docs/content.zh/docs/connectors/datastream/kinesis.md [hunk 1]
```diff
@@ -126,6 +126,45 @@ shard IDs are not consecutive (as result of dynamic re-sharding in Kinesis).
...
+The `KinesisDeserializationSchema` allows users to specify such a schema. The `T deserialize(byte[] recordValue, String partitionKey, String seqNum, long approxArrivalTimestamp, String stream, String shardId)` 
+method gets called for each Kinesis record.
+
+For convenience, Flink provides the following schemas out of the box:
+  
+1. `TypeInformationSerializationSchema` (and `TypeInformationKeyValueSerializationSchema`) which creates
+    a schema based on a Flink's `TypeInformation`. This is useful if the data is both written and read by Flink.
+    This schema is a performant Flink-specific alternative to other generic serialization approaches.
+
+2. `JsonDeserializationSchema` (and `JSONKeyValueDeserializationSchema`) which turns the serialized JSON
```
Comment (dannycranmer) [inline docs/content.zh/docs/connectors/datastream/kinesis.md:141]: I cannot see a `JsonDeserializationSchema` in the codebase, and `JSONKeyValueDeserializationSchema implements KafkaDeserializationSchema`, not sure this is right for Kinesis.

File: docs/content.zh/docs/connectors/datastream/kinesis.md [hunk 2]
```diff
@@ -126,6 +126,45 @@ shard IDs are not consecutive (as result of dynamic re-sharding in Kinesis).
...
+    into an ObjectNode object, from which fields can be accessed using `objectNode.get("field").as(Int/String/...)()`.
+    The KeyValue objectNode contains a "key" and "value" field which contain all fields, as well as
+    an optional "metadata" field that exposes the offset/partition/topic for this message.
+    
+3. `AvroDeserializationSchema` which reads data serialized with Avro format using a statically provided schema. It can
+    infer the schema from Avro generated classes (`AvroDeserializationSchema.forSpecific(...)`) or it can work with `GenericRecords`
+    with a manually provided schema (with `AvroDeserializationSchema.forGeneric(...)`). This deserialization schema expects that
+    the serialized records DO NOT contain embedded schema.
+
+    - You can use AWS's implementation of a schema registry [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html)
```
Comment (dannycranmer) [inline docs/content.zh/docs/connectors/datastream/kinesis.md:151]: nit: reword `You can use AWS's implementation of a schema registry [AWS Glue Schema Registry]` to `You can use [AWS Glue Schema Registry]`

File: docs/content/docs/connectors/datastream/kafka.md [hunk 3]
```diff
@@ -100,6 +100,12 @@ For convenience, Flink provides the following schemas out of the box:
       [Confluent Schema Registry](https://docs.confluent.io/current/schema-registry/docs/index.html). Using these deserialization schema
       record will be read with the schema that was retrieved from Schema Registry and transformed to a statically provided( either through 
       `ConfluentRegistryAvroDeserializationSchema.forGeneric(...)` or `ConfluentRegistryAvroDeserializationSchema.forSpecific(...)`).
+      
+    - You can also use AWS's implementation of a schema registry [AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html)
```
Comment (dannycranmer) [inline docs/content/docs/connectors/datastream/kafka.md:104]: nit: same as above, reword to `You can use [AWS Glue Schema Registry]`

File: flink-formats/flink-avro-glue-schema-registry/pom.xml [hunk 4]
```diff
@@ -33,7 +33,7 @@ under the License.
 	<packaging>jar</packaging>
 
 	<properties>
-		<glue.schema.registry.version>1.0.1</glue.schema.registry.version>
+		<glue.schema.registry.version>1.0.2</glue.schema.registry.version>
```
Comment (dannycranmer) [inline flink-formats/flink-avro-glue-schema-registry/pom.xml:36]: Really this change (and below) should be in a dedicated PR. Can you please raise a separate PR with a dedicated Jira issue and we should be able to merge this right away.

**Review by dannycranmer — CHANGES_REQUESTED** [review]

**Review by dannycranmer — COMMENTED** [review]
LGTM

**Pull request merged**